Content.Shared/Chemistry/Reagent/ReagentId.cs

```diff
@@ -75,7 +75,21 @@ public override bool Equals(object? obj)
 
     public override int GetHashCode()
     {
-        return HashCode.Combine(Prototype, Data);
+        // We need to make we take the hash code of Data by value in order for
+        // .Except to work properly.
+        var hash = 17;
+        unchecked
+        {
+            if (Data?.Count != 0)
+            {
+                foreach (var data in Data ?? [])
+                {
+                    hash = hash * 23 + data.GetHashCode();
```

Comment:
Oh I also still need to make sure that changing this to hash by value is probably not going to break anything.